build.sbt

```diff
@@ -17,6 +17,8 @@ ThisBuild / tlBaseVersion := "0.11"
 
 ThisBuild / crossScalaVersions := Seq(Scala213, Scala212)
 ThisBuild / scalaVersion := Scala212
+ThisBuild / tlSkipIrrelevantScalas := true
+ThisBuild / githubWorkflowArtifactUpload := false // doesn't work with scoverage
```

Comment:
Do you know if anyone has tried using the snapshots? I don't think they were working.

Reply:
@armanbilge i have one project that depends on snapshots of a specific spark version (31), let me double check it though.

Reply:
@armanbilge I can confirm, current snapshots properly work 🚀 .

Reply:
Thanks for trying! When you say current snapshot, which hash did you try? If they were working then we don't need to change this setting.

Reply:
@armanbilge [0.11.1-22-df6dc10-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/org/typelevel/frameless-core_2.12/0.11.1-22-df6dc10-SNAPSHOT/)

Reply:
@pomadchin Thanks! Huh. I want to think about why it works here first (I had some problem with other projects using coverage) but yeah it would be good if we can remove this setting, it reduces the efficiency actually.

Reply:
@armanbilge gotcha, do you want me to create a PR or you have some time? (since you're already pushing it)

Reply:
I'll make a PR to remove it again, after I convince myself that it's okay 😆

Reply:
Thanks! 💪

Reply:
@pomadchin so I looked into this, and I was able to understand why it was working :)

During the build jobs, it is actually compiling the code _twice_: first, with coverage enabled, for the tests, then a second time, with coverage disabled, for the MiMa binary-compatibility check. The second artifacts were the ones being published, which is why there was no problem. I was afraid that the first artifacts compiled with coverage enabled were being published, which would be bad 😅 

Now that I understand this, actually I think it's safer to keep the changes I made here. I'm concerned that if the MiMa step is skipped (for example, because you are publishing a new breaking version of frameless with no previous versions to compare binary-compatibility with) then it will not do the second compile and it will publish the artifacts with coverage enabled.

In the current setup, the artifacts are compiled again during the publish job (instead of being downloaded from cache). So it is less efficient, but it only adds a few extra minutes to the publish job, which is not too bad.

Another alternative could be to run the coverage check as a one extra job. Some projects such as cats-parse and skunk do it like this. However, I think it only makes sense if you are only checking coverage for one Scala/Spark version. If you want to check coverage for all the combinations it's better to keep it as it is, otherwise we will double the number of jobs in the matrix 😆